src/firewall/nft.rs

```diff
@@ -847,6 +710,55 @@ impl firewall::FirewallDriver for Nftables {
     }
 }
 
+// compare two rules, we only check the chain name and expr,
+// while we can do rule1 == rule2 it will not work how we like.
+// As we use this to compare rules from nft against rules created
+// by us in memory it means the handle id and index can never match.
+fn cmp_rules(rule1: &schema::Rule, rule2: &schema::Rule) -> bool {
+    if rule1.chain == rule2.chain && rule1.expr == rule2.expr {
```

Comment:
This is so much easier than I thought it would have been. Would have made life so much easier when I wrote the code if I knew.

Reply:
yeah it also opens up the door for some other potential problems, we can never alter the rules we create as teardown would fail to removes rules created on previous versions. But well that is already the case on iptables and if we ever have to do that kind of thing we can add some extra rule matchers to handle old rules as well.